### PR TITLE
Implement --filename option for kops delete

### LINKED ISSUE #2071: kops delete -f
We have `create -f` ... you know we should have `delete -f` as well!

### PR DESCRIPTION
**Pull request by MrTrustor**
Works for both clusters and instancegroups. Mostly inspired from the code in the kops create command.

I am wondering if I should implement the ``--yes`` option. If we want to be close to the kubectl behaviour, maybe not.

This is totally untested for now.

Fixes #2071

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/2240)

### CODE REVIEW

**Comment by kris-nova** [comment]
@k8s-boy okay to test

**Comment by MrTrustor** [comment]
After some basic testing, this seems to work as expected.

File: cmd/kops/delete.go [hunk 1]
```diff
@@ -17,29 +17,117 @@ limitations under the License.
...
+				options.ClusterName = v.ObjectMeta.Name
+				options.Yes = d.Yes
+				err = RunDeleteCluster(factory, out, options)
+				if err != nil {
+					exitWithError(err)
+				}
+				if d.Yes {
+					fmt.Fprintf(&sb, "Deleted cluster/%s\n", v.ObjectMeta.Name)
+				}
+			case *kopsapi.InstanceGroup:
```
Comment (chrislovecnm) [inline cmd/kops/delete.go]: What if the instance groups are not included and the cluster is included? What is the behavior?

Comment (MrTrustor) [inline cmd/kops/delete.go]: Using ``kops delete -f`` on a file referencing only a cluster will call the same code as ``kops delete cluster <clustername>``, so it deletes all resources (instance groups included) linked to the cluster.

Comment (chrislovecnm) [inline cmd/kops/delete.go]: awesome

**Comment by chrislovecnm** [comment]
@k8s-bot ok to test

**Pull request merged**